fe/fe-core/src/main/java/com/starrocks/sql/optimizer/GroupExpression.java

```diff
@@ -52,6 +52,7 @@ public class GroupExpression {
     private final List<Group> inputs;
     private final Operator op;
     private final BitSet ruleMasks = new BitSet(RuleType.NUM_RULES.ordinal() + 1);
+    private final BitSet appliedRuleMasks = new BitSet(RuleType.NUM_RULES.ordinal() + 1);
```

Comment:
```
Add appliedRuleMasks  to mark this GroupExpression is derived from which rule.
```

why need this add appliedRuleMasks?

Reply:
Use this to track whether this GroupExpression is derived from MV rules or not. So then can be used in `ForceRewrite` later.